Summary field in ms teams

vaibhav-db opened the pull request.

abhishekdahiya-db committed:
SYMPHONYP-1135-summary-feature

abhishekdahiya-db committed:
SYMPHONYP-1135-summary-feature

vaibhav-db committed:
SYMPHONYP-1135 code revert

abhishekdahiya-db committed:
SYMPHONYP-1135-summary-feature

vaibhav-db committed:
Merge branch 'spring-bot-master-db' of https://stash.gto.intranet.db.com:8082/scm/symphonyp/symphony-java-toolkit into feature/SYMPHONYP-1135-summary-feature

robmoffat reviewed (commented):
maybe add to an existing test

abhishekdahiya-db committed:
SYMPHONYP-1135-summary-feature

robmoffat commented on libs/chat-workflow/src/main/java/org/finos/springbot/workflow/response/templating/TeamsWorkResponse.java:
```diff
@@ -0,0 +1,32 @@
+import java.util.Map;
+
+public class TeamsWorkResponse extends WorkResponse {
+    private final String summary;
```
You shouldn't need to add a field to the TeamsWorkResponse - instead, just declare the name of a key and then add it to the `data` map.

robmoffat replied:
Also, this is breaking the separation of concerns in spring-bot, as you've added a Teams-Specific class  to chat-workflow.  Everything teams-related should go in the teams project.  

I think you'll see based on what I've written already that this class is unnecessary

vaibhav-db replied:
Yes, right.. we will move to team module

robmoffat commented on libs/teams/teams-chat-workflow-spring-boot-starter/src/main/java/org/finos/springbot/teams/handlers/TeamsResponseHandler.java:
```diff
@@ -171,7 +164,8 @@ private BiFunction<? super ResourceResponse, Throwable, ResourceResponse> handle
 						wr.getData().put(AdaptiveCardTemplateProvider.FORMID_KEY, "just-buttons");
 						JsonNode expandedJson = workTemplater.applyTemplate(buttonsJson, wr);
-						return sendCardResponse(expandedJson, (TeamsAddressable) wr.getAddress(), wr.getData()).get();
+						Optional<String> responseSummaryOptional = wr instanceof TeamsWorkResponse ? Optional.ofNullable(((TeamsWorkResponse) wr).getSummary()) : Optional.empty() ;
```
Seems like an unnecessary use of Optional here - surely the fact that it could be null is enough?

robmoffat commented:
Have a look at `AdaptiveCardTemplateProvider`.FORMID_KEY - this is an example of how the summary should be done.

vaibhav-db committed:
SYMPHONYP-1135 summary-feature

vaibhav-db committed:
Pull request #59: SYMPHONYP-1135 summary-feature

Merge in SYMPHONYP/symphony-java-toolkit from feature/SYMPHONYP-1135-summary-feature to spring-bot-master-db

* commit 'c999623ce17bfeabbada419a2b938043bc6cc3c9':
  SYMPHONYP-1135 summary-feature

vaibhav-db committed:
SYMPHONYP-1135 summary field in workresponse

vaibhav-db committed:
Pull request #60: SYMPHONYP-1135 summary field in workresponse

Merge in SYMPHONYP/symphony-java-toolkit from feature/SYMPHONYP-1135-summary-feature to spring-bot-master-db

* commit '580710a949ac2c366075c0cb1db5c16141e0fb86':
  SYMPHONYP-1135 summary field in workresponse
